### ISSUE TITLE
Spanish Translation

### ISSUE THREAD

**Issue by EA4ALX** [rogerclarkmelbourne/radioddity_gd-77_cps2.0.5#36]
Hi all, I have been working on Spanish translation of Comunnity into Spanish.
I enclose Spanish.xml and Spanish.chm to be merged,

I have decompiled de CHM file and got all the html file to translate too and this will be my next step and publish in a few days.
I suppose there will be some some way to compile the chm again, but I do not know exactly how to do  it. Any help?
Regards...

[Spanish.zip](https://github.com/rogerclarkmelbourne/radioddity_gd-77_cps2.0.5/files/1756378/Spanish.zip)

**Comment by SQ9MDD** [comment]
Check this out: https://www.microsoft.com/en-us/download/details.aspx?id=21138
You can decompile *.chm file translate and compile again.
I can do this for you... except for Spanish translation ofcourse ;)

**Comment by rogerclarkmelbourne** [comment]
Thanks I'll add the Spanish files to the development branch.

BTW. I will probably merge the development branch into the master branch and publish a new exe today (my timezone ...   Monday 26 Feb)

**Comment by rogerclarkmelbourne** [comment]
I've updated the development branch to include these new Spanish files, and also added the Polish files to the Visual Studio project and also the InnoSetup installer script, and I've ran the InnoSetup script to build a few development version exe

https://github.com/rogerclarkmelbourne/radioddity_gd-77_cps2.0.5/raw/development/installer/RadioddityCPS2.0.5CommunityEditionInstaller.exe

Please test and let me know if there are any problems.

**Comment by EA4ALX** [comment]
I have a new version of spanish.xml with added strings for moving up and down zones and the disclosure window. At this time no changes for help file, work in progress.
Please feel free to merge and let me know.

73 de EA4ALX Jorge
[Spanish.zip](https://github.com/rogerclarkmelbourne/radioddity_gd-77_cps2.0.5/files/1775289/Spanish.zip)

**Comment by rogerclarkmelbourne** [comment]
BTW. There was a small problem with xml, The text for 'as is' had non matching quote marks

So I have fixed this.

I good way to check an xml is to open it in a browser e.g. Firefox or possibly Chrome as they often parse XML and can indicate where there is an error

**Comment by rogerclarkmelbourne** [comment]
I've added the file to the development branch, and there is a new installer

**Comment by rogerclarkmelbourne** [comment]
BTW. I'm adding a new feature to download contacts from ham-digital.org "last heard" database.

I've added some new text to the XML files for this, but the functionality is not finished yet, so not everything is loaded from the xml file, and I may still change some things

**Comment by EA4ALX** [comment]
Thank you for your great work Roger.

I enclose new spanish.xml file with latest strings regarding contacts download translated.

Please feel free to merge and let me know.
[Spanish.zip](https://github.com/rogerclarkmelbourne/radioddity_gd-77_cps2.0.5/files/1792897/Spanish.zip)

**Comment by rogerclarkmelbourne** [comment]
Thanks 

I will merge this later today

**Comment by EA4ALX** [comment]
Hi again,

For some reason the latest version you have release does not apply the spanish translation in Internet Download in Contacts, the whole form remains in English.

Please, have a look and let me know.

73

**Comment by EA4ALX** [comment]
I am afraid you are not using the latest spanish.xml I sent 2 days ago. It is 3 posts up. Please, have a look!

**Comment by EA4ALX** [comment]
Found another minor bug, in Zones, "move up" is correctly translated but "move down" strings remains in english.

**Comment by rogerclarkmelbourne** [comment]
I seem to have some problems with the translations for the new Download Contacts screen as they are not being loaded :-(

So I will need to bug fix

**Comment by rogerclarkmelbourne** [comment]
OK

Bug is fixed. 

Its because you don't send the files via a PR. 

When I copy the file into my directory, it is not seen as a newer date and not copied into the debug folder where the exe is run from, and would also not be updated by the installer.

However I have now edited the file to fix formatting issue and this gives it a new date

Note. There is also one new button on the ContactsForm, for "Internet download"

I forgot to add that item to languages other than english.
But I have added that now, so you can translate "Internet download"

**Comment by rogerclarkmelbourne** [comment]
Ooops

Move down is still broken.

I'll investigate and fix

**Comment by rogerclarkmelbourne** [comment]
OK
Problem was in most XML file (except german)
Id of movedown was incorrect.

Its fixed now.

**Comment by EA4ALX** [comment]
Now it works! Thank you Roger

**Comment by rogerclarkmelbourne** [comment]
@EA4ALX 
OK.

One more thing...

I think the button for "Internet Download" on the DigitalContacts form needs to be translated...

**Comment by EA4ALX** [comment]
Yes, I know, I was waiting for next version to update. Could you do it form me and include it in the development branch?
You can translate Internet Download as "Descargas Internet"

TIA, 73

PD: I keep working on CHM translation. I have discovered a tool call CHMEditor at www.chmeditor.com
The demo lets you edit "on the fly" the chm file and recompile it again. It can translate via Google Translator API but I have a problem, every 10 or half a dozen pages, googles blocks my IP. Any ideas?

**Comment by rogerclarkmelbourne** [comment]
OK
I'll update that file

Re:CMH
OK. I have not tried much translation in Google translate, You could try using you mobile phone as you may get a new IP address if you turn to airplane mode then turn back on again (not sure but its worth trying)

Or is there a way to convert the whole CMH to text, and then translate all of it, but is likely to translate the tags as well as the data, but at least you would have all of it translated and can cut and paste

**Comment by rogerclarkmelbourne** [comment]
I've pushed a new version about 8 hours ago... I just forgot to post here to say I'd done it.

Its all in the development branch...

Can you uninstall and reinstall and test its working as I have changed from using setup.ini to store the settings to using the Windows Registry. Also the default installation location is in Program files .....

**Comment by rogerclarkmelbourne** [comment]
@EA4ALX @SQ9MDD @dg9vh 

Guys. I've updated the language XML files with some new texts for CPS 3.1.x support.

I think if you look in the commit history for your individual files, you can see whats changed, 
There is a new "Scan time" in the ZoneBasic form, and also new texts at the bottom of the "common" section.

About 1 week ago, I also added some other strings as part of the contacts download

Cheers

Roger
